src/app.rs

```diff
@@ -62,10 +62,19 @@ impl ApplicationConfig {
     ) -> Result<Self, ExitCode> {
         let config_paths = opts.config_paths_with_formats();
 
+        let graceful_shutdown_duration = if opts.no_graceful_shutdown_limit {
+            None
+        } else {
+            Some(Duration::from_secs(u64::from(
+                opts.graceful_shutdown_limit_secs,
+            )))
+        };
```

Comment:
The `bool` type recently gained a `then` function that can do this simply:
```suggestion
        let graceful_shutdown_duration = (!opts.no_graceful_shutdown_limit).then(||
            Duration::from_secs(u64::from(
                opts.graceful_shutdown_limit_secs,
            ))
        );
```

Reply:
🤯

Reply:
I bet there are dozens of places in the code base that could be refactored to use this.

Reply:
I'm slowly pointing it out, review by review. 😁

Reply:
Bruce is the biggest fan of this style 😆